### PR TITLE
ebeam_scan to multiprobe, galvo_scan to multiprobe updated to calcula…

### PR DESCRIPTION
**Pull request by tepals**
…te the gain (pixels per volt)

- Removed manual function from both scripts.
- get_deflection_angle now calculates a transform between voltages and pixel coordinates, from this transformation the rotation and scale (gain) is obtained.
- changed dcoffset call, because new dcoffset function does not require to pass a channel only 'x' or 'y'

### CODE REVIEW

File: scripts/ebeam_scan_to_multiprobe.py [hunk 1]
```diff
@@ -31,105 +31,79 @@
...
+    coordinates: (list of tuples)
         (x, y) center coordinates of the moved spot grids.
+    voltages: (list of tuples)
+        x and y offset applied to the deflectors.
 
     Returns
     -------
     phi: (float)
-        Deflection angle in radians.
+        Deflection angle in radians. Angle is counterclockwise.
```
Comment (Sabrina85) [inline scripts/ebeam_scan_to_multiprobe.py]: Can you also add where 0 is? In Odemis is 0 at 12 o'clock, which is not matching with the mathematical convention.

File: scripts/ebeam_scan_to_multiprobe.py [hunk 2]
```diff
@@ -31,105 +31,79 @@
...
-        # Solve the equation A*x = 0; i.e. find the null space of A. The solution is
-        # the eigenvector corresponding to the smallest singular value.
-        U, s, V = numpy.linalg.svd(A, full_matrices=False)
-        x = numpy.transpose(V[-1])
-
-        # Determine the angle of this line to the x-axis, -pi/2 < phi < pi/2
-        phi = numpy.arctan2(-x[0], x[1])
-    phi = (phi + math.pi / 2) % math.pi - math.pi / 2
-    return phi
+    t = transform.ScalingTransform()
```
Comment (Sabrina85) [inline scripts/ebeam_scan_to_multiprobe.py]: add a comment what this line does

File: scripts/ebeam_scan_to_multiprobe.py [hunk 3]
```diff
@@ -171,20 +140,19 @@ def main(args):
...
+            print("Added {:.2f}° to the scan rotation using SEM PC. Rotation is now set to: {:.2f}°".format(
+                    math.degrees(sem_rotation),
+                    math.degrees(scanner.rotation.value)))
         else:
-            print("Add {:.2f}° to the scan rotation using SEM PC. If negative"
-                  "subtract this value from the scan rotation.".format(
-                math.degrees(sem_rotation)))
+            print("Add {:.2f}° to the scan rotation using SEM PC.".format(
+                    math.degrees(sem_rotation)))
+        print("Gain of the e-beam deflectors is {} pixels per volt in x and {} pixels per volt in y".format(
```
Comment (Sabrina85) [inline scripts/ebeam_scan_to_multiprobe.py]: Maybe we should add somewhere what the definition of gain is. I find it still confusing, as I know gain as an amplification value.

File: scripts/galvo_scan_to_multiprobe.py [hunk 4]
```diff
@@ -40,56 +40,44 @@
...
+    coordinates: (list of tuples)
+        (x, y) center coordinates of the moved spot grids.
+    voltages: (list of tuples)
+        x and y offset applied to the deflectors.
 
     Returns
     -------
     phi: (float)
-        Angle between a line fitted through the coordinates and the y-axis. Angle is clockwise and in radians.
+        Deflection angle in radians. Angle is counterclockwise.
```
Comment (Sabrina85) [inline scripts/galvo_scan_to_multiprobe.py]: same here: would add the 0 pos info.

File: scripts/galvo_scan_to_multiprobe.py [hunk 5]
```diff
@@ -150,47 +132,41 @@ def get_galvo_rotation(ccd, channel=None, galvo_offset=3):
...
+        Experimentally determined offset of the DeScan-X galvo in Volt.
+    galvo_x_offset: (float)
         Experimentally determined offset of the DeScan-Y galvo in Volt.
 
     Returns
     -------
     galvo_rotation: (float)
         The clockwise angle, in radians, of the DeScan-Y galvo relative to the y-axis of the diagnostic camera.
+    scale: (tuple of floats)
+        The x and y scale factor between the voltages and coordinates, in pixels per volt.
```
Comment (Sabrina85) [inline scripts/galvo_scan_to_multiprobe.py]: Check if gain and scale are depicting the same, that only one term is used. Otherwise it gets confusing.

**Comment by tepals** [comment]
@Sabrina85 Updated, hope this clarifies the gain a bit.

**Pull request merged**